### PR TITLE
Fixed Grassy Terrain not healing + Added To-Do test results

### LINKED ISSUE #2820: #2737 Broke Grassy Terrain
It removed the call to `BattleScript_GrassyTerrainHeals`, so Grassy Terrain's healing effect has been removed and should be added back.

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
Fixes #2820. 
I didn't feel like adding more than the needed test to fix Grassy Terrain, so I added a way to have test placeholders instead.

## **Discord contact info**
AsparagusEduardo#6051

### CODE REVIEW

File: src/battle_util.c [hunk 1]
```diff
@@ -2127,15 +2127,25 @@ enum
...
+            gFieldStatuses &= ~terrainFlag;
+            TryToRevertMimicry();
+            
+            if (terrainFlag != STATUS_FIELD_GRASSY_TERRAIN)
+            {
+                gBattleCommunication[MULTISTRING_CHOOSER] = stringTableId;
+                BattleScriptExecute(BattleScript_TerrainEnds);
+            }
+        }
+        if (terrainFlag == STATUS_FIELD_GRASSY_TERRAIN)
```
Comment (ghoulslash) [inline src/battle_util.c:2145]: & STATUS_FIELD_GRASSY

**Review by ghoulslash — CHANGES_REQUESTED** [review]

**Comment by AsparagusEduardo** [comment]
Changes applied. Ready for re-review.

**Review by ghoulslash — APPROVED** [review]
fix conflicts

**Comment by AsparagusEduardo** [comment]
Conflict solved.

**Review by ghoulslash — APPROVED** [review]

**Pull request merged**